third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -1966,9 +1966,15 @@ fn check_elem_layout(ty: &Type, v: &Container) -> PartialVMResult<()> {
 
         (Type::Vector(_), Container::Vec(_)) => Ok(()),
 
-        (Type::Struct(_), Container::Vec(_))
+        (Type::Struct { index: _ }, Container::Vec(_))
```

Comment:
There's this long discussion of whether you should use the fall through pattern in a match statement. I think historically we refrained ourselves from using fall through because if we want to add a variant to an enum later, it will be easy for us to forget updating variant for the fall through argument. That's why this code looks so ugly because we are forcing ourselves to perform exhaustive match.